### ISSUE TITLE
Paddle Subscription Pause causes Webhook Error

### ISSUE THREAD

**Issue by fdietz** [pay-rails/pay#318]
Hi all!

I've successfully integrated Paddle in our rails app and saved a considerable amount of time with the pay gem version 2.6.10. Many thanks for your work on this gem!

After testing all subscription use case with a sandbox account, there seems to be only one with a small error. When pausing a subscription via the Paddle Dashboard it throws the following error in the webhook:

```
Started POST "/pay/webhooks/paddle" for 34.194.127.46 at 2021-04-28 19:21:18 +0200
Processing by Pay::Webhooks::PaddleController#create as */*
  Parameters: {"alert_id"=>"292451", "alert_name"=>"subscription_updated", "cancel_url"=>"https://sandbox-checkout.paddle.com/subscription/cancel?user=37011&subscription=97419&hash=895a2e497c371dd8d194c0c4e660eab56bd271c501f3ff771b319885ac28d49d", "checkout_id"=>"157492-chrea97e2382dd8-5de4e44075", "currency"=>"EUR", "email"=>"admin@test.de", "event_time"=>"2021-04-28 17:21:17", "linked_subscriptions"=>"", "marketing_consent"=>"0", "new_price"=>"29.00", "new_quantity"=>"1", "new_unit_price"=>"29.00", "next_bill_date"=>"2021-05-28", "old_next_bill_date"=>"2021-05-28", "old_price"=>"29.00", "old_quantity"=>"1", "old_status"=>"active", "old_subscription_plan_id"=>"10128", "old_unit_price"=>"29.00", "passthrough"=>"{\"owner_sgid\":\"BAh7CEkiCGdpZAY6BkVUSSIfZ2lkOi8vbmV3c2xldHRlci9BY2NvdW50LzEGOwBUSSIMcHVycG9zZQY7AFRJIgxkZWZhdWx0BjsAVEkiD2V4cGlyZXNfYXQGOwBUSSIdMjAyMS0wNS0yOFQxMDo0MzozMy44NzJaBjsAVA==--0db803c03d0ef10eb4ecea210b9a7f89f6b8b822\"}", "paused_at"=>"2021-04-28 17:21:16", "paused_from"=>"2021-05-28 00:00:00", "paused_reason"=>"voluntary", "status"=>"active", "subscription_id"=>"97419", "subscription_plan_id"=>"10128", "update_url"=>"https://sandbox-checkout.paddle.com/subscription/update?user=37011&subscription=97419&hash=f607f1227d1a49fed58115158a67d3505d06152d143c4993ea2a4b5b733ede59", "user_id"=>"37011", "p_signature"=>"eNmpWIsncJU6gpwF6E9Tncpw01ct3+68HfZLvHITdwRymOhTH9kEDaN8NbbIAOPgF1BrPU9RLb73eleYArjX7wqOATo4j4CnKq3T16BiK1tSpMSMGhbsRmDoK7Qy8JwiV1x2YKT+8u629OKqUyFNbRqhuQT7jjtIyqpx7XMBGOQfSBDGggkAhUsQ/EO8t25YmhSDolR5NxxiKW9nwnbzS/S4EsbHkby52hd7Kzu5UjfzDAgYWBdRqRIPVPCq5BjL+4GiirbEFWVWEYaM80mkzNrldoMBhg3WGUjNanRRRC0glX7x/vWe5/BKByT/TPVH2ra8sMbL2POuO5vC/6UUfeOoTfI4k8RcdI9TZJwWC/uMjM4EBdArVQ5Vrp8B++Thvtd59aqQTRBZeMHx8beXKH0EpYNLu17KjAWMX3o0jYUjs7UR/k5qSwhl1YYzngz8t8giP8JAtXx2r17BBYNGqa2PpxBkOrl4mpBaA7tAcGIkahwPu645cJkbv6cYUHYGuTOomXaV7zz87z5gfWfOCHm4GNXz8S5dWYJriVWDDPraa5SrqxgmKzDJmo1J7P2miXwWzcSTgpft5y1v4H6yxir1r/jTfi+hQXTcPYNjoyPzrEXKY6Q1YWLq/X3Q30DIG3755Ll34vUn9VG/AsR2HRAi9u92kr1JO5MrWapavHE="}
  Pay::Subscription Load (0.2ms)  SELECT "pay_subscriptions".* FROM "pay_subscriptions" WHERE "pay_subscriptions"."processor" = $1 AND "pay_subscriptions"."processor_id" = $2 LIMIT $3  [["processor", "paddle"], ["processor_id", "97419"], ["LIMIT", 1]]
Completed 500 Internal Server Error in 5ms (ActiveRecord: 0.2ms | Allocations: 2422)

IndexError (string not matched):
  
activerecord (6.1.3.1) lib/active_record/store.rb:230:in `[]='
activerecord (6.1.3.1) lib/active_record/store.rb:230:in `write'
activerecord (6.1.3.1) lib/active_record/store.rb:245:in `write'
activerecord (6.1.3.1) lib/active_record/store.rb:213:in `write_store_attribute'
activerecord (6.1.3.1) lib/active_record/store.rb:136:in `block (3 levels) in store_accessor'
pay (2.6.10) lib/pay/paddle/webhooks/subscription_updated.rb:19:in `call'
pay (2.6.10) lib/pay/webhooks/delegator.rb:46:in `call'
activesupport (6.1.3.1) lib/active_support/notifications/fanout.rb:186:in `fini
```

In this line it sets the `paddle_paused_from` attribute on the `subscription` model.

Note, that when calling `subscription.pause` (and `resume`) in the rails console instead it works correctly. This only happens in the webhook.

Has anyone else run into the same issue? Please, let me know if you need more information to further debug the issue. 

Many thanks in advance!
-Frederik

**Comment by excid3** [comment]
What column type is `pay_data` on your Pay::Subscription table?

**Comment by fdietz** [comment]
I guess, you are refering to the `data` column in the `pay_subscriptions` table? In my case it is using `jsonb` since I'm running postgres?

**Comment by excid3** [comment]
Yep. Thanks. 👍 

What's the value of that column? Almost sounds like it's a string not a hash.

**Comment by fdietz** [comment]
Sadly, I don't have the original value anymore. I've tested various different ways and in the process also deleted and recreated subscriptions.

But, here's the good news: I've just started all over again with a fresh account and subscription and was able to manually do all scenarios including pausing/resuming subscriptions successfully :tada: The current value of the `data` column is a hash.

I'm so sorry for the false alarm! Might be that I've played around too much in my testing and caused the erroneous state of the `data` entry.

Will have to do some further testing, but it looks like I cannot reproduce the setup from yesterday. Thanks a lot for your assistance :-) I will have a closer look at the `data` column in my test to check if something wrong is going on. Thanks for pointing that out. Will close this issue.
